apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/pagerules/PageRuleFinderServiceImpl.java

```diff
@@ -0,0 +1,85 @@
+package gov.cdc.nbs.questionbank.pagerules;
+
+import gov.cdc.nbs.questionbank.entity.pagerule.WaRuleMetadata;
+import gov.cdc.nbs.questionbank.model.ViewRuleResponse;
+import gov.cdc.nbs.questionbank.pagerules.repository.WaRuleMetaDataRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Slf4j
+@Service
+public class PageRuleFinderServiceImpl implements PageRuleFinderService {
+
+    private final WaRuleMetaDataRepository waRuleMetaDataRepository;
+
+    public PageRuleFinderServiceImpl(
+            WaRuleMetaDataRepository waRuleMetaDataRepository) {
+        this.waRuleMetaDataRepository = waRuleMetaDataRepository;
+    }
+
+    @Override
+    public ViewRuleResponse getRuleResponse(Long ruleId) {
+        WaRuleMetadata ruleMetadata = waRuleMetaDataRepository.getReferenceById(ruleId);
+        List<String> sourceValues = new ArrayList<>();
+        List<String> targetValues = new ArrayList<>();
+        if (ruleMetadata.getSourceValues() == null || ruleMetadata.getTargetQuestionIdentifier() == null) {
+            sourceValues.add(null);
+            targetValues.add(null);
```

Comment:
Is there a reason to add `null` to the list instead of just having the list be empty?